### PR TITLE
[FEATURE] New Dataset Support: Microsoft SQL Server

### PR DESCRIPTION
**Pull request by alexsherstinsky**
This Pull Request contains the implementation of the Microsoft SQL Server as a new Dataset for Great Expectations.

Specific Enhancements:
- Microsoft SQL Server SQLLite Connection String
- Work around via temporary table for "Error 130: Cannot perform an aggregate function on an expression containing an aggregate or a subquery" -- this is a safe (single-table) and transactional (create and populate) implementation.
- Using "LIKE" and "NOT LIKE" pattern matching (since there is no Regular Expressions capability in Microsoft SQL Server; this requires further discussion (e.g., allow the same "expect_column_values_to__regex<_type>" expectation to remain, but document the API to explain how it will behave in the special case of the Microsoft SQL Server).
- Introducing docker-compose.yml in order to be able to test against a local instance of Microsoft SQL Server
- Solutions for schema differences:
   - "BOOLEAN" --> "BIT"
   - Strings with NULL values -> "NVARCHAR"
   - Floating point "Infinity" -> 1.79e308
- Tests have been updated and pass.

Notable Omission: No CLI yet; it will be added upon the merging of the present pull request. In the mean time, use the current CLI interface and enter the connection string at the prompt for the "Other SQLAlchemy Engine" option:

```
"mssql+pyodbc://user:password@host:port/database?driver=ODBC Driver 17 for SQL Server&charset=utf8&autocommit=true"
```

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations.
After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

**Comment by Aylr** [comment]
This is an exciting and big PR! To split up the work of reviewing this PR, I’m going to do a thorough test of this PR branch on a real dataset in the wild and will leave a detailed code review for someone else.

I’ll verify use of everything in the description and these open MSSQL issues: https://github.com/great-expectations/great_expectations/issues?q=is%3Aissue+is%3Aopen+mssql+

File: .travis.yml [hunk 1]
```diff
@@ -43,13 +43,16 @@ jobs:
...
+#    - stage: test
+#      dist: xenial
+#      before_install:
+#        - sudo apt-get update
+#        - sudo apt-get install build-essential libssl-dev libffi-dev g++ python3-dev unixodbc-dev python-pip python-pyodbc
+#      install:
+#        - pip install --no-cache-dir --requirement requirements-dev.txt
+#      script:
+#        - isort --check-only --skip docs/
+#        - black --check --exclude docs/ .
```
Comment (Aylr) [inline .travis.yml:55]: Is this intentional?

Comment (alexsherstinsky) [reply .travis.yml:55]: Yes -- per @jcampbell

File: requirements-dev-build.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# A lot of Great Expectations' codebase and tests are tied to specific databases, cloud providers, etc.
+# In this file, comments on each row indicate which tests that package is used for.
+# To run tests for smaller subsets of infrastructure, please look at other requirements-dev-*.txt files.
+# Otherwise (i.e., if/when you are not concerned with running tests), please ignore these comments.
```
Comment (Aylr) [inline requirements-dev-build.txt:4]: ❤️ ❤️

File: tests/test_utils.py [hunk 3]
```diff
@@ -172,7 +222,8 @@ def get_dataset(
         return PandasDataset(df, profiler=profiler, caching=caching)
 
     elif dataset_type == "sqlite":
-        from sqlalchemy import create_engine
+        if not create_engine:
```
Comment (Aylr) [inline tests/test_utils.py:225]: ❤️

File: .travis.yml (hunk 1 above)
Comment (Aylr) [reply .travis.yml:55]: Answered: Yes.

**Review by Aylr — CHANGES_REQUESTED** [review]
Pending another acceptance test for latest temp table error. We are getting so close!

**Comment by Aylr** [comment]
FYI I'm actively giving this another round of acceptance testing on a big MSSQL server this morning.

**Review by Aylr — APPROVED** [review]

**Pull request merged**